lib/views/chatroom/chatroom_components/chat_bubble.dart

```diff
@@ -0,0 +1,367 @@
+                            const SizedBox(height: 8),
+                            Text(
+                              _time,
+                              style: GoogleFonts.roboto(
+                                fontSize: 12,
+                                color: Colors.black.withOpacity(0.6),
+                              ),
+                            ),
+                          ],
+                        ),
+                      ),
+                    ),
+                  ),
+                ),
+              ),
+              const SizedBox(width: 6),
+              _isSent
+                  ? Container(
+                      width: 42,
+                      height: 42,
+                      decoration: BoxDecoration(
+                        color: Colors.blue,
+                        borderRadius: BorderRadius.circular(21),
+                        image: _profileImageUrl != null
+                            ? DecorationImage(
+                                image: NetworkImage(_profileImageUrl),
+                                fit: BoxFit.cover,
+                              )
+                            : null,
+                      ),
+                    )
+                  : const SizedBox(),
+            ],
+          ),
+        ),
+        if (reactions.isNotEmpty)
+          Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 64.0),
+            child: Row(
+              mainAxisAlignment:
+                  _isSent ? MainAxisAlignment.end : MainAxisAlignment.start,
+              children: reactions.map((e) {
+                switch (e) {
+                  case Reactions.like:
+                    return ReactionChip(
+                      text: "1",
+                      icon: emojiParser.emojify(":thumbsup:"),
+                    );
+                  case Reactions.dislike:
+                    return ReactionChip(
+                      text: "1",
+                      icon: emojiParser.emojify(":thumbsdown:"),
+                    );
+                  case Reactions.love:
+                    return ReactionChip(
+                      text: "1",
+                      icon: emojiParser.emojify(":heart:"),
+                    );
+                  case Reactions.happy:
+                    return ReactionChip(
+                      text: "1",
+                      icon: emojiParser.emojify(":joy:"),
+                    );
+                  case Reactions.sad:
+                    return ReactionChip(
+                      text: "1",
+                      icon: emojiParser.emojify(":cry:"),
+                    );
+                  default:
+                    return const SizedBox();
+                }
+              }).toList(),
+            ),
+          ),
+      ],
+    );
+  }
+
+  void refresh() {
+    setState(() {});
+  }
+
+  void selectMessage(String messageId) {}
+
+  Future<Widget> getContent(
```

Comment:
It will be better to create a factory class for getting the content so that we can directly use the content_factory widget instead of this function. You can have a look here https://github.com/NateshR/LikeMinds-Flutter-Feed-SX/pull/4/files#diff-52d4422aebcafa0730937a36a76b0ed3bea703393aff2977a0d2f770c5ca7852. This follows the factory design pattern.

Reply:
Will sort this out when I get back to this with the data coming in